src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import itertools
+import logging
+import os
+from pathlib import Path
+from typing import Any, Iterable, Mapping, Optional
+
+import toml
+from packaging.version import InvalidVersion, Version
+from pkg_resources import Requirement
+
+from pants.base.build_environment import get_buildroot
+
+logger = logging.getLogger(__name__)
+
+
+def get_max_caret(proj_name: str, version: str) -> str:
+    major = "0"
+    minor = "0"
+    micro = "0"
+
+    try:
+        parsed_version = Version(version)
+    except InvalidVersion:
+        logger.warning(
+            f"Warning: version {version} for {proj_name} is not PEP440-compliant; this requirement"
+            f" will be left as >={version},<{version}"
+        )
+        return version
```

Comment:
> \>={version},<{version}

This is impossible to ever be resolved. 

```
pex 'ansicolors>=1.1.8,<1.1.8'
ERROR: Could not find a version that satisfies the requirement ansicolors<1.1.8,>=1.1.8 (from versions: 1.0, 1.0.1, 1.0.2, 1.1.5, 1.1.6, 1.1.7, 1.1.8)
ERROR: No matching distribution found for ansicolors<1.1.8,>=1.1.8
```

Instead, for now, I think the best thing to do is error. You could say something similar to your errors below:

> Failed to parse requirement `{req}` in {path} loaded by the `poetry_requirements` macro: {err}\n\nIf you believe this requirement is valid, consider opening an issue at https://github.com/pantsbuild/pants/issues so that we can update Pants's Poetry macro to support this.